Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Create/CreateStreetcodeHandler.cs

```diff
@@ -59,6 +59,12 @@ public async Task<Result<int>> Handle(CreateStreetcodeCommand request, Cancellat
             {
                 var streetcode = StreetcodeFactory.CreateStreetcode(request.Streetcode.StreetcodeType);
                 _mapper.Map(request.Streetcode, streetcode);
+                if (streetcode.Index < 1 || streetcode.Index > 9999)
```

Comment:
Thank you for your feedback. What is the best place to place constants( that replace magic number )? Handler, Entity or something else?

Reply:
There's no official rule for this, because it depends on the context. If it is a number that simply describes a certain condition when an action can be executed or not (especially if the given number is reused in many methods/tests of the class), then they are usually declared in that specific class to ensure visibility to other methods. If it directly concerns the validation of some field, then it should be placed in the validator (e.g. FluentValidations validator). _Read about validation at the end of the comment._ When there are much more of these constants and they apply to the same context, a static class may be created where they are grouped, then these constants are not spread over all files.

Taking into account all the circumstances of this particular project (which also concerns validation (or its absence)), you can place them in the Handler for now.

In my opinion, the backend of this project suffers from insufficient validation or its absence in many places (not only DB entities). I opened [a ticket](https://github.com/ita-social-projects/StreetCode/issues/308) for this a long time ago, but it seems that this task has not been completed yet.